lib/src/component_declaration/component_base.dart

```diff
@@ -433,7 +461,7 @@ abstract class UiStatefulComponent<TProps extends UiProps, TState extends UiStat
 ///
 /// > Note: Implements MapViewMixin instead of extending it so that the abstract state declarations
 /// don't need a constructor. The generated implementations can mix that functionality in.
-abstract class UiState extends Object implements StateMapViewMixin, MapViewMixin, Map {
+abstract class UiState extends MapBase implements StateMapViewMixin, MapViewMixin, Map {
```

Comment:
`MapBase` seems to work to get us fully compatible with both Dart 1 and Dart 2 at the same time, so I'm fine with this until we've fully switched to Dart 2.

At that point, we should:
- stop extending from `MapBase`, or at least add in overrides that directly proxy the underlying map's members, as opposed to using `MapBase's` potentially less efficient versions.
- hopefully be able to simplify all this map stuff and remove all the weird DDC workarounds

I believe `UiProps` needs the same treatment, since it needs to implement all of the same new Dart2 map members. I think there might be an analyzer bug where unimplemented members don't show up; but you should be able to see if it's needed by doing something like `Dom.div().entries` and see if it's implemented.

Reply:
@greglittlefield-wf I have removed the `MapBase` bits [in my WIP branch for dart 2 changes (AF-233)](https://github.com/Workiva/over_react/commit/42d4aeb81e1de6c946973be7df407588bc3877e2#diff-7c38c4a4b204fe0e8293ccc6b5c4a1b6R462)